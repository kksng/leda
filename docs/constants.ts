export enum Urls {
  Concept = '/',
  Api = '/api',
  ApiEvents = '/api/events',
  Markup = '/markup',
  Customization = '/customization',
  CustomizationStyles = '/customization/styles',
  CustomizationClasses = '/customization/classes',
  CustomizationElements = '/customization/elements',
  FormsDifferences = '/forms/differences',
  ValidationDifferences = '/validation/differences',
}
