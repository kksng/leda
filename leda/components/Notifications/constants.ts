export const DEFAULT_NOTIFICATION_LIFETIME = 5000; // milliseconds

export enum ChangeMethods {
  Delay = 'delay',
  CloseIconClick = 'close-icon-click',
}
