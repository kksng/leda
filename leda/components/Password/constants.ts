export const DEFAULT_MIN_PASSWORD_EVALUATION_LENGTH = 4;

export enum PasswordStrength {
  Low = 'low',
  Medium = 'medium',
  Strong = 'strong',
}
