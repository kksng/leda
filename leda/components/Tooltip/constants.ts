export const defaultArrowSize = 7;
export const defaultPosition = 'top';
export const defaultTransitionTimeout = 200;
