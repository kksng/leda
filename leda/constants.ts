export const NO_DATA_TEMPLATES = {
  default: 'default' as const,
};

export const FILE_SIZE_UNITS = {
  byte: 'byte' as const,
  kB: 'kB' as const,
  MB: 'MB' as const,
  GB: 'GB' as const,
  TB: 'TB' as const,
};

export const FILE_SIZE_RUS_UNITS = {
  [FILE_SIZE_UNITS.byte]: 'байт' as const,
  [FILE_SIZE_UNITS.kB]: 'Кб' as const,
  [FILE_SIZE_UNITS.MB]: 'Мб' as const,
  [FILE_SIZE_UNITS.GB]: 'Гб' as const,
  [FILE_SIZE_UNITS.TB]: 'Тб' as const,
};

export const MAX_FILE_SIZE = 100 * 1024 * 1024;
export const MIN_FILE_SIZE = 0;

export const COMPONENTS_NAMESPACES = {
  autoComplete: 'autoComplete',
  button: 'button',
  buttonGroup: 'buttonGroup',
  checkBox: 'checkBox',
  collapse: 'collapse',
  collapseBody: 'collapseBody',
  collapseHeading: 'collapseHeading',
  collapsePanel: 'collapsePanel',
  currency: 'currency',
  dateTimeInput: 'dateTimeInput',
  dateTimeInputRange: 'dateTimeInputRange',
  dropDown: 'dropDown',
  dropDownLink: 'dropDownLink',
  dropDownSelect: 'dropDownSelect',
  dropZone: 'dropZone',
  fileDrop: 'fileDrop',
  fileUpload: 'fileUpload',
  input: 'input',
  loader: 'loader',
  maskedInput: 'maskedInput',
  modal: 'modal',
  modalBody: 'modalBody',
  modalFooter: 'modalFooter',
  modalHeader: 'modalHeader',
  multiSelect: 'multiSelect',
  notifications: 'notifications',
  numericRange: 'numericRange',
  numericTextBox: 'numericTextBox',
  pagination: 'pagination',
  password: 'password',
  progressBar: 'progressBar',
  radio: 'radio',
  rating: 'rating',
  slider: 'slider',
  statusBar: 'statusBar',
  stickyPanel: 'stickyPanel',
  suggestionList: 'suggestionList',
  switcher: 'switcher',
  tabs: 'tabs',
  tags: 'tags',
  textarea: 'textarea',
  tooltip: 'tooltip',
  vstepper: 'vstepper',
} as const;

/** Ошибки DropZone, FileDrop и FileUpload */
export const ERROR_MESSAGES = [
  { errorCode: 1, message: 'Файл меньше допустимого размера' },
  { errorCode: 2, message: 'Превышен максимальный размер файла' },
  { errorCode: 3, message: 'Недопустимый формат файла' },
  { errorCode: 4, message: 'Файл уже загружен' },
  { errorCode: 5, message: 'Превышено максимальное количество файлов' },
  { errorCode: 6, message: 'Превышена максимальная длина имени файла' },
];
