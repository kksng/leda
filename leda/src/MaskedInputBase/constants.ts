import { MaskRules } from './types';

export const baseMaskRules: MaskRules = {
  '#': {
    validate: (char: string) => /\d/.test(char),
  },
  l: {
    validate: (char: string) => /[a-zA-Z]/.test(char),
  },
  L: {
    validate: (char: string) => /[a-zA-Z]/.test(char),
    transform: (char: string) => char.toUpperCase(),
  },
  c: {
    validate: (char: string) => /[а-яА-Я]/.test(char),
  },
  C: {
    validate: (char: string) => /[а-яА-Я]/.test(char),
    transform: (char: string) => char.toUpperCase(),
  },
  x: {
    validate: () => true,
  },
  w: {
    validate: (char: string) => /[0-9a-zA-Z]/.test(char),
  },
  z: {
    validate: (char: string) => /[0-9a-zA-Zа-яА-Я]/.test(char),
  },
};

export enum INPUT_METHODS {
  add = 'add',
  remove = 'remove',
  replace = 'replace',
  // метод, обозначающий отсутствие изменений
  nothing = 'nothing'
}

export const DEFAULT_PLACEHOLDER_CHAR = '_';
