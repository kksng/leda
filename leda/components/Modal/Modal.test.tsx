// @ts-nocheck
import React from 'react';
import { mount } from 'enzyme';
import { Modal, ModalBody, ModalHeader } from './index';
import { WINDOW_SIZES } from './constants';

describe('Modal SNAPSHOTS', () => {
  let wrapper;
  afterEach(() => { wrapper.unmount(); });

  it('should render', () => {
    wrapper = mount(<Modal isOpen />);

    expect(wrapper).toMatchSnapshot();
  });

  describe('different component states', () => {
    it('should be closed', () => {
      wrapper = mount(<Modal isOpen={false} />);

      expect(wrapper).toMatchSnapshot();
    });

    it('should have header', () => {
      const modal = (
        <Modal isOpen>
          <ModalHeader>
            <h1>Bread all over the Head</h1>
          </ModalHeader>
          <ModalBody>
            <span>Yo, buddy</span>
          </ModalBody>
        </Modal>
      );
      wrapper = mount(modal);

      expect(wrapper.find('div.modal-header')).toHaveLength(1);

      expect(wrapper.find('h1').text()).toEqual('Bread all over the Head');

      expect(wrapper).toMatchSnapshot();
    });

    it('should render closing icon in wrapper', () => {
      const modal = (
        <Modal isOpen onCloseButtonClick={jest.fn()}>
          <ModalHeader>
            <h1>Bread all over the Head</h1>
          </ModalHeader>
          <ModalBody>
            <span>Yo, buddy</span>
          </ModalBody>
        </Modal>
      );
      wrapper = mount(modal);

      expect(wrapper.find('A')).toHaveLength(1);

      // проверим, что иконка действительно в ModalWindow
      expect(wrapper.find('.modal-window a.modal-cross')).toHaveLength(1);

      expect(wrapper).toMatchSnapshot();
    });
  });
});

describe('Modal HANDLERS', () => {
  it.skip('should trigger onOverlayClick outside of modal', () => {
    // TODO: Обработчик вызывается при ev.target === ev.currentTarget, а enzyme этого не может
    const onClickHandler = jest.fn();
    const wrapper = mount(<Modal isOpen onOverlayClick={onClickHandler} />);

    wrapper.find('.modal-wrapper').props().onClick();

    expect(onClickHandler).toHaveBeenCalled();
  });

  it('should trigger onCloseIconClick', () => {
    const onClickHandler = jest.fn();
    const modal = (
      <Modal isOpen onCloseButtonClick={onClickHandler}>
        <ModalBody>
          <span>Yo, buddy</span>
        </ModalBody>
      </Modal>
    );
    const wrapper = mount(modal);

    wrapper.find('a.modal-cross').props().onClick({ preventDefault: () => true });

    expect(onClickHandler).toHaveBeenCalled();

    wrapper.unmount();
  });

  it('should delete overflow:hidden style when unmount', () => {
    const modal = (
      <Modal isOpen={false}>
        <ModalBody>
          <span>Yo, buddy</span>
        </ModalBody>
      </Modal>
    );

    const wrapper = mount(modal);

    wrapper.setProps({ isOpen: true });

    wrapper.update();

    expect(document.body.style.overflow).toBe('hidden');

    wrapper.unmount();

    expect(document.body.style.overflow).toBe('');
  });
});

describe('Modal ATTRIBUTES', () => {
  it('should have className, change classes through props and className should not change prop-classes', () => {
    const wrapper = mount(<Modal _box isOpen><div className="test">default</div></Modal>);

    expect(wrapper.find('div.modal-wrapper').hasClass('box')).toBeTruthy();


    wrapper.setProps({ _active: true, _box: false });

    expect(wrapper.find('div.modal-wrapper').hasClass('box')).toBeFalsy();

    expect(wrapper.find('div.modal-wrapper').hasClass('active')).toBeTruthy();


    wrapper.setProps({ className: 'testClass' });

    expect(wrapper.find('div.modal-wrapper').hasClass('box')).toBeFalsy();

    expect(wrapper.find('div.modal-wrapper').hasClass('active')).toBeTruthy();

    expect(wrapper.find('div.modal-wrapper').hasClass('testClass')).toBeTruthy();
  });

  it('should have children prop', () => {
    const modal = (
      <Modal isOpen>
        <ModalHeader>
          <h1>Bread all over the Head</h1>
        </ModalHeader>
        <ModalBody>
          <div className="lvl1">
            <span className="lvl2">TEXT</span>
          </div>
        </ModalBody>
      </Modal>
    );
    const wrapper = mount(modal);

    const modalBody = wrapper.find('ModalBody');

    expect(modalBody.contains(<div className="lvl1"><span className="lvl2">TEXT</span></div>)).toBeTruthy();
  });

  it('should be opened if isOpen and vice versa', () => {
    const modal = (
      <Modal isOpen>
        <ModalHeader>
          <h1>Bread all over the Head</h1>
        </ModalHeader>
        <ModalBody>
          Body
        </ModalBody>
      </Modal>
    );
    const wrapper = mount(modal);

    const modalBody = wrapper.find('ModalBody');

    expect(modalBody.contains('Body')).toBeTruthy();

    wrapper.setProps({ isOpen: false });

    expect(wrapper.find('ModalBody')).toHaveLength(0);
  });

  describe('size', () => {
    it('should accept sm', () => {
      const modal = (
        <Modal isOpen size="sm">
          <ModalHeader>
            <h1>Bread all over the Head</h1>
          </ModalHeader>
          <ModalBody>
            Body
          </ModalBody>
        </Modal>
      );
      const wrapper = mount(modal);

      expect(wrapper.find('div.modal-window').props().style.width).toEqual(WINDOW_SIZES.sm);
    });

    it('should accept md', () => {
      const modal = (
        <Modal isOpen size="md">
          <ModalHeader>
            <h1>Bread all over the Head</h1>
          </ModalHeader>
          <ModalBody>
            Body
          </ModalBody>
        </Modal>
      );
      const wrapper = mount(modal);

      expect(wrapper.find('div.modal-window').props().style.width).toEqual(WINDOW_SIZES.md);
    });

    it('should accept lg', () => {
      const modal = (
        <Modal isOpen size="lg">
          <ModalHeader>
            <h1>Bread all over the Head</h1>
          </ModalHeader>
          <ModalBody>
            Body
          </ModalBody>
        </Modal>
      );
      const wrapper = mount(modal);

      expect(wrapper.find('div.modal-window').props().style.width).toEqual(WINDOW_SIZES.lg);
    });
  });
});
