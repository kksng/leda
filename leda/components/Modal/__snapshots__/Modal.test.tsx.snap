// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal SNAPSHOTS different component states should be closed 1`] = `
<Modal
  isOpen={false}
/>
`;

exports[`Modal SNAPSHOTS different component states should have header 1`] = `
<Modal
  isOpen={true}
>
  <Portal
    containerInfo={
      <body
        style="overflow: hidden;"
      >
        <div
          class="modal-wrapper"
        >
          <div
            class="modal-window"
            style="width: 600px;"
          >
            <a
              class="modal-cross"
            />
            <div
              class="modal-header"
            >
              <h1>
                Bread all over the Head
              </h1>
            </div>
            <div
              class="modal-body"
            >
              <span>
                Yo, buddy
              </span>
            </div>
          </div>
        </div>
      </body>
    }
  >
    <Component
      innerRef={null}
      isOpen={true}
    >
      <Wrapper
        className="modal-wrapper"
        onClick={[Function]}
      >
        <Div
          className="modal-wrapper"
          onClick={[Function]}
        >
          <div
            className="modal-wrapper"
            onClick={[Function]}
          >
            <div
              className="modal-window"
              style={
                Object {
                  "width": 600,
                }
              }
            >
              <Icon
                className="modal-cross"
                onClick={[Function]}
              >
                <A
                  className="modal-cross"
                  onClick={[Function]}
                >
                  <a
                    className="modal-cross"
                    onClick={[Function]}
                  />
                </A>
              </Icon>
              <ModalHeader>
                <Wrapper
                  className="modal-header"
                >
                  <Div
                    className="modal-header"
                  >
                    <div
                      className="modal-header"
                    >
                      <h1>
                        Bread all over the Head
                      </h1>
                    </div>
                  </Div>
                </Wrapper>
              </ModalHeader>
              <ModalBody>
                <Wrapper
                  className="modal-body"
                >
                  <Div
                    className="modal-body"
                  >
                    <div
                      className="modal-body"
                    >
                      <span>
                        Yo, buddy
                      </span>
                    </div>
                  </Div>
                </Wrapper>
              </ModalBody>
            </div>
          </div>
        </Div>
      </Wrapper>
    </Component>
  </Portal>
</Modal>
`;

exports[`Modal SNAPSHOTS different component states should render closing icon in wrapper 1`] = `
<Modal
  isOpen={true}
  onCloseButtonClick={[MockFunction]}
>
  <Portal
    containerInfo={
      <body
        style="overflow: hidden;"
      >
        <div
          class="modal-wrapper"
        >
          <div
            class="modal-window"
            style="width: 600px;"
          >
            <a
              class="modal-cross"
            />
            <div
              class="modal-header"
            >
              <h1>
                Bread all over the Head
              </h1>
            </div>
            <div
              class="modal-body"
            >
              <span>
                Yo, buddy
              </span>
            </div>
          </div>
        </div>
      </body>
    }
  >
    <Component
      innerRef={null}
      isOpen={true}
      onCloseButtonClick={[MockFunction]}
    >
      <Wrapper
        className="modal-wrapper"
        onClick={[Function]}
      >
        <Div
          className="modal-wrapper"
          onClick={[Function]}
        >
          <div
            className="modal-wrapper"
            onClick={[Function]}
          >
            <div
              className="modal-window"
              style={
                Object {
                  "width": 600,
                }
              }
            >
              <Icon
                className="modal-cross"
                onClick={[Function]}
              >
                <A
                  className="modal-cross"
                  onClick={[Function]}
                >
                  <a
                    className="modal-cross"
                    onClick={[Function]}
                  />
                </A>
              </Icon>
              <ModalHeader>
                <Wrapper
                  className="modal-header"
                >
                  <Div
                    className="modal-header"
                  >
                    <div
                      className="modal-header"
                    >
                      <h1>
                        Bread all over the Head
                      </h1>
                    </div>
                  </Div>
                </Wrapper>
              </ModalHeader>
              <ModalBody>
                <Wrapper
                  className="modal-body"
                >
                  <Div
                    className="modal-body"
                  >
                    <div
                      className="modal-body"
                    >
                      <span>
                        Yo, buddy
                      </span>
                    </div>
                  </Div>
                </Wrapper>
              </ModalBody>
            </div>
          </div>
        </Div>
      </Wrapper>
    </Component>
  </Portal>
</Modal>
`;

exports[`Modal SNAPSHOTS should render 1`] = `
<Modal
  isOpen={true}
>
  <Portal
    containerInfo={
      <body
        style="overflow: hidden;"
      >
        <div
          class="modal-wrapper"
        >
          <div
            class="modal-window"
            style="width: 600px;"
          >
            <a
              class="modal-cross"
            />
          </div>
        </div>
      </body>
    }
  >
    <Component
      innerRef={null}
      isOpen={true}
    >
      <Wrapper
        className="modal-wrapper"
        onClick={[Function]}
      >
        <Div
          className="modal-wrapper"
          onClick={[Function]}
        >
          <div
            className="modal-wrapper"
            onClick={[Function]}
          >
            <div
              className="modal-window"
              style={
                Object {
                  "width": 600,
                }
              }
            >
              <Icon
                className="modal-cross"
                onClick={[Function]}
              >
                <A
                  className="modal-cross"
                  onClick={[Function]}
                >
                  <a
                    className="modal-cross"
                    onClick={[Function]}
                  />
                </A>
              </Icon>
            </div>
          </div>
        </Div>
      </Wrapper>
    </Component>
  </Portal>
</Modal>
`;
